src/Domain/LeanCode.TimeProvider/Time.cs

```diff
@@ -2,10 +2,12 @@ namespace LeanCode.TimeProvider;
 
 public static class Time
 {
-    private static ITimeProvider timeProvider = new UtcTimeProvider();
+    private static readonly AsyncLocal<System.TimeProvider?> Provider = new();
 
-    public static DateTime Now => timeProvider.Now;
-    public static DateTimeOffset NowWithOffset => timeProvider.NowWithOffset;
+    public static System.TimeProvider CurrentProvider => Provider.Value ?? System.TimeProvider.System;
 
-    public static void Use(ITimeProvider newProvider) => timeProvider = newProvider;
+    public static DateTime Now => NowWithOffset.UtcDateTime;
+    public static DateTimeOffset NowWithOffset => CurrentProvider.GetUtcNow();
```

Comment:
This should not force you to UTC. Same with `Now`. The provider should define if you get +00:00 or any other offset.

Reply:
In case of `DateTime` I disagree, the only kind of timestamp that it makes sense to return is UTC one, everything else is worthless. In case of `DateTimeOffset`, sure, I can swap this to call `GetLocalNow` instead. However, do we want the default implementation to return non-UTC timestamps or should I write a `TimeProvider` that pretends to be in `+00:00` timezone for our purposes?

Reply:
Datetime has two kinds, and we should not force one or the other. There are use cases where you do want it to be local (although you would ignore the kind, you would not convert to UTC), and not converted to UTC (we haven't used that in a long time, but sometimes you have to). That's the very reason why time provider exists in the first place - to not force you to go UTC/non-UTC in the first place.

We should have a UTC provider by default, and system one should not be used.

Reply:
> The way it is now, it's still possible to retrieve local DateTime by calling Time.NowWithOffset.DateTime, I think UTC timestamps are still the primary use case for DateTime and so I'd prefer Now to always return in UTC. At least when you actually want local timestamp, it will be more obvious to see what you're doing and why.

That's not really the case. The main idea is to allow changing the effective timezone _globally_, in a single place (e.g. setting env variable with the timezone), without the code knowing that it is working in a different timezone. Forcing you to use UTC _all the time_ defies the point of this abstraction a little. The main usecase is to have `Time.Now` and `Time.NowWithOffset` behave the same, i.e. if one returns zoned datetime, the other should also return _the same time_. And you should be able to switch between the behavior without changing the code.

 > In fact, we used to have a test that did

We did not. This code tests `FixedTimeProvider`, not general time provider. In tests that works, but that is the only place where the simplification makes sense.